core/meterer/meterer.go

```diff
@@ -47,23 +42,23 @@ func NewMeterer(
 	return &Meterer{
 		Config: config,
 
-		ChainState:    paymentChainState,
-		OffchainStore: offchainStore,
+		ChainPaymentState: paymentChainState,
+		OffchainStore:     offchainStore,
 
 		logger: logger.With("component", "Meterer"),
 	}
 }
 
 // Start starts to periodically refreshing the on-chain state
-func (m *Meterer) Start(ctx context.Context) {
+func (m *Meterer) Start(ctx context.Context, updateInterval time.Duration) {
 	go func() {
-		ticker := time.NewTicker(1 * time.Hour)
+		ticker := time.NewTicker(updateInterval)
```

Comment:
nit: would `updateInterval` be ever different from `m.UpdateInterval`? Should we just use `m.UpdateInterval` instead of passing it in as a param?

Reply:
thanks for pointing it out! updated